h/services/group_members.py

```diff
@@ -24,6 +24,29 @@ def __init__(self, db, user_fetcher, publish):
         self.user_fetcher = user_fetcher
         self.publish = publish
 
+    def get(self, group: Group, roles: list[GroupMembershipRoles] | None = None):
+        """
+        Return `group`'s memberships.
+
+        If `roles` is None return all of `group`'s memberships.
+
+        If `roles` is not None return only those memberships matching the given role(s).
+
+        If multiple roles are given return all memberships matching *any* of
+        the given roles.
+        """
+        query = select(GroupMembership).where(GroupMembership.group == group)
+
+        if roles:
+            if len(roles) == 1:
```

Comment:
Is this necessary? Test seem to pass without it, looks like or_ with just one element renders to just the condition (without OR).

https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.or_

Reply:
:+1: I checked this and if we just allow the query to contain an `or_()` with only one predicate in it SQLAlchemy compiles that into an SQL query without an `OR`. So yes we can remove this